pkg/api/validation/validation.go

```diff
@@ -2784,6 +2788,18 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	return allErrs
 }
 
+// Validate that opaque integer resource quantities in a resource list are
+// indeed integers.
+func ValidateOpaqueIntQuants(rs api.ResourceList, fldPath *field.Path) field.ErrorList {
```

Comment:
ValidateResourceQuantityValue in validation.go does something similar.  I would use that same pattern and just move integer checking into a shared call